### PR TITLE
Increase visibility of allowed failures

### PR DESCRIPTION
**Pull request by keewis**
Right now, allowed failures are hard to see, so this tries to add warnings to increase the visibility without removing the `allow_failure` flag.

 - [x] related to #4574

### CODE REVIEW

**Review by max-sixty — APPROVED** [review]
👍

**Comment by keewis** [comment]
It seems we can set the result to "succeeded with issues", which works pretty well in the pipeline overview. However, the only difference this makes on github is that it prints the result in the checks tab – no fancy icon, github seems to only know the `success`, `failure`, `cancelled`, and `skipped` results, so we still get a green check mark.

We still need to decide whether to set `allow_failure: true` for the `upstream-dev` CI. I'd vote for waiting until the summary bar at the bottom of a PR (which prints the number of failed and successful checks) also lists the number of partially successful checks.

File: ci/azure/unit-tests.yml [hunk 1]
```diff
@@ -15,7 +15,12 @@ steps:
     --junitxml=junit/test-results.xml \
     --cov=xarray \
     --cov-report=xml \
-    $(pytest_extra_flags) || [ "$ALLOW_FAILURE" = "true" ]
+    $(pytest_extra_flags) \
+    || ( \
+      [ "$ALLOW_FAILURE" = "true" ] \
+      && echo -e "\043#vso[task.logissue type=warning]Ignored CI failure!!" \
```
Comment (mathause) [inline ci/azure/unit-tests.yml:21]:
```suggestion
      && echo -e "\043#vso[task.logissue type=warning]Allowed CI failure!!" \
```

**Review by mathause — COMMENTED** [review]
Thanks for figuring this out. Re the `upstream-dev` - I think if #4583 works as it should we could also allow failures for the upstream dev.

**Comment by keewis** [comment]
so people can see the difference after I removed the intentionally failing test, this is how a failed CI with `allowed_failure: true` looks like: 

<details><summary>the pipelines overview</summary>

warning

</details>
<details><summary>checks tab (<tt>pydata.xarray</tt>)</summary>

overview_total

</details>

<details><summary>checks tab (<tt>pydata.xarray (Linux py38-flaky)</tt>)</summary

overview_flaky

</details>

**Comment by keewis** [comment]
if we agree on keeping the `allow_failure` flag on `py38-upstream-dev` this should be ready for merging.

**Comment by max-sixty** [comment]
Great! Definitely LGTM!

**Comment by keewis** [comment]
should we merge this? I think we can decide which CI to keep (pipelines or github actions) in a different PR / issue

**Comment by keewis** [comment]
we have two votes agreeing with me, so let's merge!

**Pull request merged**